R/treeview.R

```diff
@@ -1,394 +1,463 @@
-			a = shared_anc2[ which.min( e0$ndesc[shared_anc2] ) ]
-			sc0$tr2mrca[ sc0$node_number == a ] <- inode
+	for (i in seq_along(stres2)){
+		inode <- i + ape::Ntip(tr2)
+		uv <- stats::na.omit(sc0$node_number[match(stres2[[i]]$tip.label,
+		                                    sc0$representative)])
+		shared_anc <- Reduce(intersect,
+		                     e0$ancestors[uv])
+		shared_anc2 <- setdiff(intersect(shared_anc,
+		                                 sc0$node_number),
+		                       uv)
+		if (length(shared_anc2) > 0) {
+			a <- shared_anc2[which.min(e0$ndesc[shared_anc2])]
+			sc0$tr2mrca[sc0$node_number == a] <- inode
 		}
 	}
 	## tips (takes precedence if overlap in tr2mrca )
-	i <- which( sc0$representative %in% tr2$tip.label )
-	sc0$tr2mrca[i] <- match( sc0$representative[i], tr2$tip.label )
+	i <- which(sc0$representative %in% tr2$tip.label)
+	sc0$tr2mrca[i] <- match(sc0$representative[i],
+	                        tr2$tip.label)
 
 	# tree data frames
 	## tips
-	sc2 <- sc0[ !is.na( sc0$tr2mrca ) , ]
-	sc2$date_range <- sapply( 1:nrow( sc2 ) , function(i) glue( '{sc2$least_recent_tip[i]} -> {sc2$most_recent_tip[i]}') )
-	tdvars = unique( c(branch_cols, 'logistic_growth_rate', 'clock_outlier', 'cluster_size', 'date_range', 'cluster_id', 'region_summary', 'cocirc_lineage_summary', 'lineage' , 'tr2mrca')  )
-	td0 <- sc2[ sc2$tr2mrca <= Ntip( tr2 )
-	 , tdvars
-	]
-	td0$lineages = td0$lineage
-	td0$cocirc_summary = td0$cocirc_lineage_summary
-	td0$node = td0$tr2mrca
-	td0$internal = 'N'
+	sc2 <- sc0[!is.na(sc0$tr2mrca), ]
+	sc2$date_range <- sapply(seq_len(nrow(sc2)),
+	                         function(i) glue::glue("{sc2$least_recent_tip[i]} -> {sc2$most_recent_tip[i]}"))
+	tdvars <- unique(c(branch_cols,
+	                 "logistic_growth_rate",
+	                 "clock_outlier",
+	                 "cluster_size",
+	                 "date_range",
+	                 "cluster_id",
+	                 "region_summary",
+	                 "cocirc_lineage_summary",
+	                 "lineage",
+	                 "tr2mrca"))
+	td0 <- sc2[sc2$tr2mrca <= ape::Ntip(tr2), tdvars]
+	td0$lineages <- td0$lineage
+	td0$cocirc_summary <- td0$cocirc_lineage_summary
+	td0$node <- td0$tr2mrca
+	td0$internal <- "N"
 	##internal
-	td1 <- sc2[ sc2$tr2mrca > Ntip( tr2 )
-	, tdvars
-	]
-	if ( nrow( td1 ) > 0 ){
-		td1$lineages = td1$lineage
-		td1$cocirc_summary = td1$cocirc_lineage_summary
-		td1$node = td1$tr2mrca
-		td1$internal = 'Y'
+	td1 <- sc2[sc2$tr2mrca > ape::Ntip(tr2), tdvars]
+	if (nrow(td1) > 0) {
+		td1$lineages <- td1$lineage
+		td1$cocirc_summary <- td1$cocirc_lineage_summary
+		td1$node <- td1$tr2mrca
+		td1$internal <- "Y"
 		td1$cluster_size <- 0
-		x = setdiff( (Ntip(tr2)+1):(Ntip(tr2) + Nnode(tr2)) ,   td1$node ) # make sure every node represented
-		td1 = merge( td1, data.frame( node = x ), all =TRUE )
-		td <- rbind( td0, td1 )
-	} else{
-		td = td0
+		x <- setdiff((ape::Ntip(tr2) + 1):(ape::Ntip(tr2) + ape::Nnode(tr2)),
```

Comment:
```suggestion
		x <- setdiff(ape::Ntip(tr2) + seq_len(ape::Nnode(tr2)),
```

Reply:
Is this the same thing? The +1 is part of the previous bracket

Reply:
It think it should be,

- (x + 1) : (x + y) should create the sequence [x+1, x+2, ..., x+y]
 - and x + seq_len(y) creates x + [1, 2, ..., y] = [x + 1, x+2, ..., x+y]

I'm happy to rescind my change. So long as Nnode(tr2) > 1 there should be no surprises.